src/SnippetProvider/RaygunSnippetProvider.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace SilverStripe\TagManager\SnippetProvider;
+
+use SilverStripe\Core\Environment;
+use SilverStripe\TagManager\SnippetProvider;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms;
+use SilverStripe\Core\Manifest\ModuleResourceLoader;
+
+/**
+ * A snippet provider that lets you add Raygun
+ */
+class RaygunSnippetProvider implements SnippetProvider
+{
+    public function getTitle()
+    {
+        return "Raygun crash reporting (JavaScript front-end)";
+    }
+
+    public function getParamFields()
+    {
+        $imgSrc = ModuleResourceLoader::singleton()->resolveURL('sminnee/tagmanager:client/img/raygun-guide.png');
+
+        return new FieldList(
+            (new Forms\CheckboxField('ENV_APP_KEY', 'Use preinstalled "silverstripe/raygun" module'))
+                ->setDescription('Choose this if your developers have installed "silverstripe/raygun" module'),
+            (new Forms\TextField('USER_APP_KEY', 'API KEY'))->setDescription('Will look like alphanumeric password'),
+            new Forms\LiteralField('HelpImage', "<p><img src=\"$imgSrc\" style=\"width: 100%; border-radius: 30px; box-shadow: 2px 2px 20px #CCC;\"></p>")
+        );
+    }
+
+    public function getSummary(array $params)
+    {
+        return $this->getTitle();
+    }
+
+    public function getSnippets(array $params)
+    {
+        if (isset($params['ENV_APP_KEY']) && $params['ENV_APP_KEY']) {
+            $apiKey = Environment::getEnv('SS_RAYGUN_APP_KEY');
+        } elseif (isset($params['USER_APP_KEY'])) {
+            $apiKey = trim(preg_replace('/[^A-Za-z0-9_\-]/', '', $params['USER_APP_KEY']));
+        } else {
+            $apiKey = null;
+        }
+
+        $head = <<<HTML
```

Comment:
Suggest putting this in a separate file

Reply:
by a separate file do you mean a static `.js` asset, or a `.ss` template?
in first case I guess it's worth it as the block itself is itself a CDN file inclusion with an extra check it hasn't been loaded yet.
In the second case it feels like it can increase module complexity unnecessarily?

Reply:
I don't mind it as is. If we put it into its own `js` file, that's just one more thing to add to `expose`, and it means everyone who uses this module who does not use Raygun now has a raygun script sitting in their public directory.